src/goto-instrument/code_contracts.cpp

```diff
@@ -299,24 +299,16 @@ bool code_contractst::apply_contract(
   return false;
 }
 
-void code_contractst::code_contracts(
+void code_contractst::apply_loop_contract(
   goto_functionst::goto_functiont &goto_function)
 {
   local_may_aliast local_may_alias(goto_function);
   natural_loops_mutablet natural_loops(goto_function.body);
 
   // iterate over the (natural) loops in the function
-  for(natural_loops_mutablet::loop_mapt::const_iterator l_it =
-        natural_loops.loop_map.begin();
-      l_it != natural_loops.loop_map.end();
-      l_it++)
+  for(const auto &loop : natural_loops.loop_map)
     check_apply_invariants(
-      goto_function, local_may_alias, l_it->first, l_it->second);
-
-  // look at all function calls
-  Forall_goto_program_instructions(ins, goto_function.body)
-    if(ins->is_function_call())
-      apply_contract(goto_function.body, ins);
```

Comment:
Still d20bbe1

Case in point: In this case it's a bit hard for me to see why this loop was removed. It's combined with a fairly straightforward refactoring of the preceding loop so it's difficult for me see if this was an intentional change or a mistake that snuck in during refactoring.

Reply:
This particular function was perhaps the old entry point for the old contracts flag. It was used to replace loops with invariant contracts and function calls with function contracts. I removed this loop intentionally because the replacement of calls with contracts now happens in `replace_calls`. You can see a very similar loop in `replace_calls`:

https://github.com/diffblue/cbmc/blob/947fb8fc11d85c21593c3947701babef78b48b45/src/goto-instrument/code_contracts.cpp#L894--L916